core/network/subnet.go

```diff
@@ -197,6 +197,27 @@ func (s SubnetInfos) GetByCIDR(cidr string) (SubnetInfos, error) {
 	return matching, nil
 }
 
+// Get by address returns subnets that based on IP range,
+// include the input IP address.
+func (s SubnetInfos) GetByAddress(addr string) (SubnetInfos, error) {
+	ip := net.ParseIP(addr)
+	if ip == nil {
+		return nil, errors.NotValidf("%q as IP address", addr)
+	}
+
+	var subs SubnetInfos
+	for _, sub := range s {
+		_, ipNet, err := net.ParseCIDR(sub.CIDR)
```

Comment:
When you need access to the CIDR of a subnet, you can also use `sub.ParsedCIDRNetwork()` which I added some time ago. It will memoize (and auto-update when needed) the result of `net.ParseCIDR`.

Reply:
Good tip. I will amend.